kms_reencrypt/executors/kms.py

```diff
@@ -7,6 +7,9 @@ def __init__(self, client_s3, kms_key_arn: str):
         self.kms_key_arn = kms_key_arn
 
     def process(self, bucket: str, key: str):
+        if key[-1] == "/":
+            return
+
```

Comment:
Actually earlier I was pointing out to fix at the filtering level (instead of at the effecting level here):
https://github.com/dsaidgovsg/kms-reencrypt/blob/feat--process-wrongly-encrypted/app.py#L55

Basically change to:

```python
content_keys = [content_obj["Key"] for content_obj in page["Contents"] if content_obj["Key"][-1] != "/"]
```

The benefit is do once and affect both `process` I guess? Not sure what do you think?

---

The other comment would be to erm add comments to explain "empty dir" with trailing slash can end up in `Contents` especially if using AWS S3 console to create new directory, which is easier to explain if done at the filtering side as well I guess.

Reply:
Good point, then we can filter away the "empty dir" keys so they're not part of the KMS `filter` as well as `executor`, probably will remove more unwanted side effects. Edited: https://github.com/dsaidgovsg/kms-reencrypt/pull/11/commits/9500572fd09a9d871ec801cb0ab3a1db80433cd5